tests/ext-popup.test.js

```diff
@@ -23,187 +26,205 @@ function observeChange(element) {
   startBtn.click();
-  await promiseElementChanged;
+  await observeMonitorStatusChanged;
 
-  // popup after clicking "Start Monitor" button
-  // all extensions are being watched
+  // extensions are being monitored
   expect(startBtn.hasAttribute('disabled')).toBeTruthy();
   expect(stopBtn.hasAttribute('disabled')).toBeFalsy();
 
-  expect(monitorStatus.textContent).toBe('Extensions are being monitored');
+  expect(monitorStatus.textContent).toBe(MONITOR_ENABLED_MSG);
   expect(monitorStatus.classList.contains('failure')).toBeFalsy();
   expect(monitorStatus.classList.contains('success')).toBeTruthy();
 
   expect(errorText.textContent).toBe('');
-
-  getMonitorStatusFn.mockClear();
-  startMonitorFn.mockClear();
 });
 
-test('stop monitoring all the extensions from popup', async () => {
+test('clicking on stop monitoring button should display that the extensions are not being monitored', async () => {
   document.body.innerHTML = popupBody;
 
-  const getMonitorStatusFn = jest.spyOn(ExtListen, 'getMonitorStatus');
-  const stopMonitorFn = jest.spyOn(ExtListen, 'stopMonitor');
+  // assume, extensions are being monitored
+  getMonitorStatusSpyFn.mockResolvedValueOnce(true);
+  // extensions are not being monitored after the stop monitoring button is clicked
+  getMonitorStatusSpyFn.mockResolvedValueOnce(false);
+  stopMonitorSpyFn.mockResolvedValue(true);
 
   const popup = new Popup();
+  await popup.init();
 
   const startBtn = popup.startMonitorAllBtn;
   const stopBtn = popup.stopMonitorAllBtn;
   const monitorStatus = popup.monitorStatusText;
   const errorText = popup.errorMsgText;
 
-  getMonitorStatusFn.mockResolvedValueOnce(true);
-  getMonitorStatusFn.mockResolvedValueOnce(false);
-  stopMonitorFn.mockResolvedValue(true);
-
-  await popup.init();
-
-  expect(startBtn.hasAttribute('disabled')).toBeTruthy();
-  expect(stopBtn.hasAttribute('disabled')).toBeFalsy();
-
-  expect(monitorStatus.textContent).toBe('Extensions are being monitored');
-  expect(monitorStatus.classList.contains('failure')).toBeFalsy();
-  expect(monitorStatus.classList.contains('success')).toBeTruthy();
-
-  expect(errorText.textContent).toBe('');
-
-  const promiseElementChanged = observeChange(monitorStatus);
+  const observeMonitorStatusChanged = observeChange(monitorStatus);
   stopBtn.click();
-  await promiseElementChanged;
+  await observeMonitorStatusChanged;
 
   expect(startBtn.hasAttribute('disabled')).toBeFalsy();
   expect(stopBtn.hasAttribute('disabled')).toBeTruthy();
 
-  expect(monitorStatus.textContent).toBe('No extensions are being monitored');
+  expect(monitorStatus.textContent).toBe(MONITOR_DISABLED_MSG);
   expect(monitorStatus.classList.contains('failure')).toBeTruthy();
   expect(monitorStatus.classList.contains('success')).toBeFalsy();
   expect(errorText.textContent).toBe('');
+});
 
-  getMonitorStatusFn.mockClear();
+test('clicking on "view activity logs page" button should open a new tab with activity logs page', async () => {
```

Comment:
This test name is improved from `clicking on "view activity logs page" button should open the activitylog page` to `clicking on "view activity logs page" button should open a new tab with activity logs page`, which seemed more concrete result that we expect.